zhmccli/_cmd_imageprofile.py

```diff
@@ -0,0 +1,1918 @@
+    properties = options_to_properties(org_options, name_map)
+
+    if org_options['disk-partition-id'] == 'auto':
+        properties['disk-partition-id-automatic'] = True
+        # 'disk-partition-id will be unchanged
+    elif org_options['disk-partition-id'] is not None:
+        properties['disk-partition-id-automatic'] = False
+        properties['disk-partition-id'] = str2int(
+            cmd_ctx, 'disk-partition-id', org_options['disk-partition-id'])
+
+    if org_options['absolute-cp-capping'] is not None:
+        value = absolute_capping_value(
+            cmd_ctx, org_options, 'absolute-cp-capping')
+        properties['absolute-general-purpose-capping'] = value
+
+    if org_options['absolute-ifl-capping'] is not None:
+        value = absolute_capping_value(
+            cmd_ctx, org_options, 'absolute-ifl-capping')
+        properties['absolute-ifl-capping'] = value
+
+    if org_options['absolute-zaap-capping'] is not None:
+        value = absolute_capping_value(
+            cmd_ctx, org_options, 'absolute-zaap-capping')
+        properties['absolute-aap-capping'] = value
+
+    if org_options['absolute-ziip-capping'] is not None:
+        value = absolute_capping_value(
+            cmd_ctx, org_options, 'absolute-ziip-capping')
+        properties['absolute-ziip-capping'] = value
+
+    if org_options['absolute-icf-capping'] is not None:
+        value = absolute_capping_value(
+            cmd_ctx, org_options, 'absolute-icf-capping')
+        properties['absolute-icf-capping'] = value
+
+    if org_options['zaware-host-name'] == '':
+        properties['zaware-host-name'] = None
+    if org_options['zaware-master-userid'] == '':
+        properties['zaware-master-userid'] = None
+    if org_options['zaware-master-pw'] == '':
+        properties['zaware-master-pw'] = None
+
+    if org_options['ssc-host-name'] == '':
+        properties['ssc-host-name'] = None
+    if org_options['ssc-master-userid'] == '':
+        properties['ssc-master-userid'] = None
+    if org_options['ssc-master-pw'] == '':
+        properties['ssc-master-pw'] = None
+
+    if not properties:
+        cmd_ctx.spinner.stop()
+        click.echo("No properties specified for updating image activation "
+                   "profile '{ip}' for CPC '{c}'.".
+                   format(ip=imageprofile_name, c=cpc_name))
+        return
+
+    try:
+        imageprofile.update_properties(properties)
+    except zhmcclient.Error as exc:
+        raise click_exception(exc, cmd_ctx.error_format)
+
+    cmd_ctx.spinner.stop()
+    click.echo("Image activation profile '{ip}' for CPC '{c}' has been "
+               "updated.".format(ip=imageprofile_name, c=cpc_name))
+
+
+def cmd_imageprofile_create(cmd_ctx, cpc_name, options):
+    # pylint: disable=missing-function-docstring
+    # pylint: disable=unreachable
+
+    client = zhmcclient.Client(cmd_ctx.session)
+    cpc = find_cpc(cmd_ctx, client, cpc_name)
+
+    name_map = {
```

Comment:
Exactly which code?

Reply:
Well, the name_map for update is quite similar?

Reply:
Just like other code is "quite similar".

I assume you suggest that the two name_map's could be set up from a part that is defined commonly for the create and update command, and then amended with a command-specific part.

It is already difficult enough to make sure that all options defined on the commands are also handled properly, and distributing the name_map setup to different places does not make that easier, because now you are concerned with making sure that you did not fgorget any option between the common part and the command-specific parts.

I believe the criteria "here the editor shows the same line as elsewhere" is not a good criteria.

It also consumes too much energy. We are loosing sight of the really important review points, like should we copy the default profile in the cli code.

Reply:
Agreed.